iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepGrammar.java

```diff
@@ -84,20 +87,36 @@ public Expression EXPRESSION() {
           STRING_LITERAL_VALUE())));
   }
 
-  // TODO SONARIAC-934 Extending of LiteralValue in grammar should return the proper Expression implementation
-  public StringLiteral LITERAL_VALUE() {
-    return b.<StringLiteral>nonterminal(BicepLexicalGrammar.LITERAL_VALUE).is(
-      f.stringLiteral(
+  public Expression LITERAL_VALUE() {
+    return b.<Expression>nonterminal(BicepLexicalGrammar.LITERAL_VALUE).is(
+      b.firstOf(
+        NUMERIC_LITERAL(),
+        BOOLEAN_LITERAL(),
+        NULL_LITERAL()));
+  }
+
+  public NumericLiteral NUMERIC_LITERAL() {
+    return b.<NumericLiteral>nonterminal(BicepLexicalGrammar.NUMERIC_LITERAL).is(
+      f.numericLiteral(b.token(BicepLexicalGrammar.NUMERIC_LITERAL_VALUE)));
+  }
+
+  public BooleanLiteral BOOLEAN_LITERAL() {
+    return b.<BooleanLiteral>nonterminal(BicepLexicalGrammar.BOOLEAN_LITERAL).is(
+      f.booleanLiteral(
         b.firstOf(
-          b.token(BicepLexicalGrammar.NUMBER_LITERAL),
-          b.token(BicepLexicalGrammar.TRUE_LITERAL),
-          b.token(BicepLexicalGrammar.FALSE_LITERAL),
-          b.token(BicepLexicalGrammar.NULL_LITERAL))));
+          b.token(BicepLexicalGrammar.TRUE_LITERAL_VALUE),
+          b.token(BicepLexicalGrammar.FALSE_LITERAL_VALUE))));
+  }
+
+  public NullLiteral NULL_LITERAL() {
+    return b.<NullLiteral>nonterminal(BicepLexicalGrammar.NULL_LITERAL).is(
+      f.nullLiteral(
+        b.token(BicepLexicalGrammar.NULL_LITERAL_VALUE)));
   }
 
   public StringLiteral STRING_LITERAL_VALUE() {
     return b.<StringLiteral>nonterminal().is(
-      f.stringLiteral(b.token(BicepLexicalGrammar.STRING_LITERAL)));
+      f.stringLiteral(b.token(BicepLexicalGrammar.STRING_LITERAL_VALUE)));
```

Comment:
What about single quotes? Should we include them as arguments in `f.stringLiteral`?

Reply:
This part is more like temporary support of string in our `Expression`.
I expect it to be progressively removed as we further implement elements such as `expression` or `interpString`.
I added a comment in this sense.
The question you are raising should definitely be tackled at those points.